src/index.js

```diff
@@ -0,0 +1,232 @@
+    CTRL_EQUALS: {
+      description: 'increase font size',
+      action: ''
+    },
+    // File Management
+
+    CTRL_O: {
+      description: 'Open source; may just focus file explorer',
+      action: ''
+    },
+    // CURSOR_ENTER: {
+    //   description: 'keys navigate and select files',
+    //   action: ''
+    // },
+    BACKSPACE: {
+      description: 'exits file view (project) (to project view)',
+      action: ''
+    },
+    CTRL_S: {
+      description: 'Save current source code file to its current location on disk',
+      action: ''
+    },
+    CTRL_P: {
+      description: 'Print current source',
+      action: ''
+    },
+    CTRL_TAB: {
+      description: 'Switch to next open source',
+      action: ''
+    },
+    CTRL_SHIFT_TAB: {
+      description: 'Switch to previous open source',
+      action: ''
+    },
+
+    // Code
+    CTRL_I: {
+      description: 'Identify BASIC Stamp firmware.',
+      action: ''
+    },
+    F6: {
+      description: 'Identify BASIC Stamp firmware.',
+      action: ''
+    },
+    CTRL_T: {
+      description: 'Perform a syntax check on the code.',
+      action: ''
+    },
+    F7: {
+      description: 'Perform a syntax check on the code.',
+      action: ''
+    },
+    CTRL_M: {
+      description: 'Tokenize and open Memory Map display.',
+      action: ''
+    },
+    F8: {
+      description: 'Tokenize and open Memory Map display.',
+      action: ''
+    },
+    CTRL_R: {
+      description: 'Tokenize, download to BASIC Stamp, and open Debug display if necessary.',
+      action: ''
+    },
+    F9: {
+      description: 'Tokenize, download to BASIC Stamp, and open Debug display if necessary.',
+      action: ''
+    },
+    ESC: {
+      description: 'Close current dialog.',
+      action: ''
+    }
+  };
+
+  // add keyCombos from the application
+  function register(name, opt, fn){
+    let keyCombo = name;
+    let opts = opt;
+    let handler = fn;
+    if(typeof opts === 'function'){
+      handler = opts;
+      opts = {};
+    }
+
+    let description = combo[keyCombo].description;
+
+    combo[keyCombo] = {
+      description: opts.description || description || '',
+      action: handler
+    }
+  }
+
+  // remove registered combos and events when components unmount
+  function removeCombo(keyCombo){
+    delete combo[keyCombo];
+  }
+
+  function match(e){
+    let kp = '';
+    let chName = e.key || e.keyCode || e.charCode; 
+
+    if(e.altKey){
+      kp.concat('ALT_');
+    } 
+    if(e.ctrlKey){
+      kp.concat('CTRL_');
+    }
+    if(e.shiftKey){
+      kp.concat('SHIFT_');
+    }
+
+    kp.concat(normalizeChar(chName));
+    if (combo[kp]){
+      combo[kp].action();
+    }
+  }
+
+  function normalizeChar(charCode){
+    if(typeof charCode === 'string'){
+      return charCode.toUpperCase();
+    }
+    let output = vkey(charCode);
+    return codes[output] || output;
+  }
+
+  app.expose('keypress', {
+    register: register,
+    unRegister: removeCombo,
```

Comment:
camelcasing unRegister looks horrible.  also, i prefer register to return a remover